### PR TITLE
Modified NCF in DIT/MODIT also added some benchmark tests for MODIT

### PR DESCRIPTION
**Pull request by HajimeKawahara**
I noticed that [the previous version of NCF](https://github.com/HajimeKawahara/addit/blob/main/src/addit/ncf_slow.py) in DIT/MODIT makes the computation time very slow (proportional to # of lines). So, I modified it using `jax.ops.index_add`. This modification makes the code very fast for large number of the lines (as written in the original paper of DIT). 

You can test this modification by the benchmark codes in tests/benchmark:

```sh
python modit_bm_wide_test.py
``` 
or 
```sh
pytest modit_bm_each_test.py
``` 

You can also check the results (comparison with DIT. MODIT and LPF) by a sample in tests/auto:

```sh
python modit_test.py
```

Thanks!

### CODE REVIEW

**Comment by HajimeKawahara** [comment]
see also #122

**Comment by HajimeKawahara** [comment]
[d7878ad](https://github.com/HajimeKawahara/exojax/pull/121/commits/d7878ada1afb48714a6dc9f4f2827e3e3529607b) does not change the core part of modit/dit but confirmed HMC-NUTS with the current version works.
See Reverse modeling for methane using MODIT.ipynb

**Review by ykawashima — COMMENTED** [review]
Thank you. I noticed the following three things.

1. As I also left a comment for Cross Section using Discrete Integral Transform.ipynb, 
```from exojax.spec.dit import xsvector3d```
does not work. xscextor3d does not seem to exist anymore in [dit.py](https://github.com/HajimeKawahara/exojax/blob/d7878ada1afb48714a6dc9f4f2827e3e3529607b/src/exojax/spec/dit.py)

2. In [Reverse modeling.ipynb](https://github.com/HajimeKawahara/exojax/blob/d7878ada1afb48714a6dc9f4f2827e3e3529607b/examples/tutorial/Reverse%20modeling.ipynb), maybe
```mcmc = MCMC(kernel, num_warmup, num_samples)```
should be
```mcmc = MCMC(kernel, num_warmup=num_warmup, num_samples=num_samples)```?
The notebook did not work as it is at least in my environment.

3. In [Reverse modeling for methane using MODIT.ipynb](https://github.com/HajimeKawahara/exojax/blob/d7878ada1afb48714a6dc9f4f2827e3e3529607b/examples/tutorial/Reverse%20modeling%20for%20methane%20using%20MODIT.ipynb), I think
```MMR=0.0059 #mass mixing ratio```
should be
```MMR_CH4=0.0059 #mass mixing ratio```

File: examples/tutorial/Cross Section using Discrete Integral Transform.ipynb [hunk 1]
```diff
@@ -172,18 +154,18 @@
...
+   "execution_count": 9,
    "metadata": {
     "scrolled": false
    },
    "outputs": [],
    "source": [
     "#If you want to compute nu in JAX/GPU, be careful for the precision of wavenumber.\n",
     "#reduce the digit \n",
-    "from exojax.spec.dit import xsvector3D\n",
+    "from exojax.spec.dit import xsvector3d\n",
```
[the remark this answers is not in the document]
Comment (HajimeKawahara) [reply examples/tutorial/Cross Section using Discrete Integral Transform.ipynb:165]: sorry, I changed the name of the function
- xsvector3D -> xsvector (as a default)
- xsvector -> xsvector_np

**Comment by HajimeKawahara** [comment]
Thank you! I updated the codes based on the review.

> mcmc = MCMC(kernel, num_warmup=num_warmup, num_samples=num_samples)?

Yes, I also noticed newer environments require these explicit options.

**Pull request merged**